libbeat/filter/condition.go

```diff
@@ -0,0 +1,325 @@
+package filter
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Condition struct {
+	Equals   map[string]EqualsValue
+	Contains map[string]string
+	Regexp   map[string]*regexp.Regexp
+	Range    map[string]RangeValue
+}
+
+func NewCondition(config ConditionConfig) (*Condition, error) {
+
+	c := Condition{}
+	c.Equals = map[string]EqualsValue{}
+	c.Contains = map[string]string{}
+	c.Regexp = map[string]*regexp.Regexp{}
+	c.Range = map[string]RangeValue{}
+
+	if err := c.AddEquals(config.Equals); err != nil {
+		return nil, err
+	}
+	if err := c.AddContains(config.Contains); err != nil {
+		return nil, err
+	}
+	if err := c.AddRegexp(config.Regexp); err != nil {
+		return nil, err
+	}
+	if err := c.AddRange(config.Range); err != nil {
+		return nil, err
+	}
+
+	return &c, nil
+}
+
+func (c *Condition) AddEquals(equals map[string]string) error {
+
+	for field, value := range equals {
+
+		i, err := strconv.Atoi(value)
```

Comment:
As far as I understand, the current implementation for equal supports `strings` and `int`. In case someone wants to use float it will return an error.